### PR TITLE
[core] Don't split TStorage::UpdateIsOnHeap declaration and definition

### PR DESCRIPTION
**Pull request by guitargeek**
This is done to avoid this compiler warning that I get when compiling
ROOT with gcc:

```
core/base/inc/TStorage.h:131:37: warning: declaration of ‘static void TStorage::UpdateIsOnHeap(const volatile UInt_t&, volatile UInt_t&)’ with attribute ‘noinline’ follows inline declaration [-Wattributes]
  131 | R__NEVER_INLINE void TStorage::UpdateIsOnHeap(volatile const UInt_t &uniqueID, volatile UInt_t &bits) {
      |                                     ^~~~~~~~
core/base/inc/TStorage.h:91:20: note: previous declaration of ‘static void TStorage::UpdateIsOnHeap(const volatile UInt_t&, volatile UInt_t&)’ here
   91 |    static void UpdateIsOnHeap(volatile const UInt_t &uniqueID, volatile UInt_t &bits);
```

### CODE REVIEW

**Comment by pcanal** [comment]
Which version of gcc 'starts' having this message?

File: core/base/inc/TStorage.h [hunk 1]
```diff
@@ -88,7 +88,23 @@ class TStorage {
    static Bool_t IsOnHeap(void *p);
 
    static Bool_t FilledByObjectAlloc(volatile const UInt_t* const member);
-   static void UpdateIsOnHeap(volatile const UInt_t &uniqueID, volatile UInt_t &bits);
```
Comment (pcanal) [inline core/base/inc/TStorage.h:91]: Question, does using 'just' also silence the warning (or does it fail to compile)?
```
< static void UpdateIsOnHeap(volatile const UInt_t &uniqueID, volatile UInt_t &bits);
> static R__NEVER_INLINE void UpdateIsOnHeap(volatile const UInt_t &uniqueID, volatile UInt_t &bits);

**Comment by hahnjo** [comment]
I think for `valgrind` and `valgrind-root.supp` it's important to have the definition of `TStorage::UpdateIsOnHeap` in the library where the compiler *cannot* inline it into tons of places...

**Comment by guitargeek** [comment]
Ok I see that this change is controversial :) The warning went away with a compiler update on my system, so I guess it's all good and the PR can be closed.

**Pull request closed without merging**